### ISSUE TITLE
AI aircraft twitches when flying in formation

### ISSUE THREAD

**Issue by aalbertmate** [till213/SkyDolly#86]
**Describe the bug**
AI aircraft twitches when I get closer in formation, while replaying my flight track.

**To Reproduce**
1. Recorded the flight
2. Formation
3. Replay
4. Fly WITH formation

**Expected behavior**
The AI should have a smooth movement instead of lagging and twitching.

**Application version:**

 - Version ["Flamboyant Fokker" (2023.02)
Version 0.15.0 (67cb9955)
Thu Feb 2 17:44:06 2023]

Thank you!
![FlightSimulator 2023-02-05 12-50-49-457](https://user-images.githubusercontent.com/124502440/216817627-aa3ee9d5-1d97-47c4-a385-dd4a5d21775d.jpg)

**Comment by till213** [comment]
Yes, this is - unfortunately - known as "aircraft jitter" (since the FS X day even) for AI aircraft, and documented in the "known issues".

The good news: I have some idea (based on some older SimConnect developer forum entries of the "old days" (< MSFS) how to improve AI aircraft replay. Whether it'll work (also with MSFS) we only know once implemented.

I change this into an "enhancement" ticket, as it requires me to write additional code (and the existing code is technically not "defect": it just doesn't work as well - inside MSFS - as expected).

**Comment by Orishek** [comment]
Any news about this _issue_? Planes jittering makes close-formation flying terrible :(

**Comment by till213** [comment]
Not as far as the AI aircraft jitter is concerned, no. However I have just restarted with implementation of the upcoming Sky Dolly v0.17 - as a general remark.

I am sorry for the delay, but "life has happened" recently.

**Comment by till213** [comment]
Oh and I took great interest in the Airshow Assistant:

https://msfs.touching.cloud/mods/airshowassistant/

The controlled AI aircraft follow smoothly the user aircraft. The difference: this addon works as "in process" addon, that is it runs "within MSFS" itself (but also using the SimConnect API, as it seems).

This could give such in-process addons an advantage in that they are really getting the latest data and are reliably invoked for every "simulated frame" (while Sky Dolly also reacts to the "simulated frame" event there is no guarantee that MSFS will also react in time to the subsequent SimConnect requests - there is "no service guarantee" ;)).
